### ISSUE TITLE
'Pick or create case' pop-up triggering should be consistent across the app 

### ISSUE THREAD

**Issue by adinaflorea9** [hzi-braunschweig/SORMAS-Project#11524]
### Problem Description

The 'Pick or create' pop-up is triggered inconsistently currently across the directories of SORMAS.

'Persons' directory - person page - the pop-up for 'Pick or create case' is triggered only after the user clicks 'Save' on the 'Create new case' form (implemented as per comment https://github.com/hzi-braunschweig/SORMAS-Project/issues/6053#issuecomment-1342307560);

'Contacts' directory - contact page - the pop-up for 'Pick or create case' is triggered only when the user clicks 'Create a case for this contact person';
after the user clicks 'Save' in the 'Create new case' form;

'Events' directory - event participants page - the pop-up for 'Pick or create case' is triggered the first time after the user clicks 'Create' and a second time after the user click's 'Save' in the 'Create new case' form.

### Proposed Change

The triggering of the 'Pick or create case' pop-up should be consistent across the app.

1. Should the pop-up be triggered only after the user clicked to 'Save' the 'Create new case form' - how it has been implemented for the 'Person' ages?

2. Should the pop-up only be triggered right when the user clicks on the button to create a new case - how it is implemented for 'Contact' pages?

### Additional Information

The argument that was used when the implementation was done for how the 'Pick or create' pop-ups are triggered on person pages was based on the fact that the  user can already see in the cards the entities that already exist (therefore sort of already aware of potential duplicates), so the pop-up should be triggered only after the information was inputted in the 'Create new entity' form so that it can be compared with the existing one in the system.

**Comment by MateStrysewske** [comment]
@adinaflorea9 I think the reason for this "inconsistency" is that we're dealing with different levels of information here. When creating a case for a person from the person page, we e.g. don't know about the disease or the potential report date or jurisdiction yet, so we can't really do a proper detection of duplicates.

However, when creating cases out of contacts or event participants, we have a lot more information. We most likely know the disease, and we also have a reference date and a jurisdiction that we can use in order to narrow down the list of potential duplicates (e.g. by using the report date and responsible jurisdiction of a contact or event participant). That's why we can already look for potential duplicates before presenting the user the case creation form, which is a very good idea in my eyes because we can likely spare them the hassle of typing in all the data only to be told "hey, by the way, there's probably a case that matches yours" (note that in a real world scenario, if the system detects a potential duplicate, it's most likely an actual duplicate).

So I think it makes a lot of sense to have different workflows here.

Now, for contacts and event participants it also makes sense to do another check after the user has saved the creation form because they might end up typing in a totally different report date, or they might select a different jurisdiction, in which case the check after saving the creation form can bring up different results than the check before. I think it's very unlikely that a user will end up with two redundant duplicate popups here; either they'll already choose the duplicate from the first popup, or they will end up with different results in the second one because they've e.g. chosen a different jurisdiction, in which case we NEED to display another popup anyway.

Long story short, I think it's fine the way it works right now. :)

**Comment by adinaflorea9** [comment]
@MateStrysewske Thank you for explaining the logic behind the different workflows. I agree that it does make sense to have it this way. I will discard the ticket.
